toucan_connectors/aircall/aircall_connector.py

```diff
@@ -31,42 +86,75 @@ class AircallConnector(ToucanConnector):
     bearer_auth_id: str
 
-    def _get_page_data(
-        self, endpoint, query, jq_filter: str, page_number: int, per_page: int
-    ) -> Tuple[List[dict], bool]:
-        """Get the data for a single page and the information if the page is the last one"""
-        page_raw_data = self.bearer_oauth_get_endpoint(
-            endpoint, {**query, 'per_page': per_page, 'page': page_number}
-        )
-        try:
-            is_last_page = page_raw_data['meta']['next_page_link'] is None
-        except KeyError:
-            is_last_page = True
-        page_data = pyjq.first(jq_filter, page_raw_data)
-        if isinstance(page_data, dict):
-            page_data = [page_data]
-        return page_data, is_last_page
+    async def _get_data(self, dataset: str, limit) -> Tuple[List[dict], List[dict]]:
+        """Triggers fetches for data and does preliminary filtering process"""
+        headers = {'Authorization': BEARER_API_KEY, 'Bearer-Auth-Id': self.bearer_auth_id}
+        async with ClientSession(headers=headers) as session:
+            team_data, variable_data = await asyncio.gather(
+                fetch_page('teams', [], session, limit, 0,),
+                fetch_page(dataset, [], session, limit, 0,),
+            )
+
+            team_response_list = []
+            variable_response_list = []
+            if len(team_data) > 0:
+                for data in team_data:
+                    for team_obj in data.get('teams'):
+                        team_response_list += DICTIONARY_OF_FORMATTERS.get('teams')(team_obj)
 
-    def _retrieve_data(self, data_source: AircallDataSource) -> pd.DataFrame:
-        endpoint = nosql_apply_parameters_to_query(data_source.endpoint, data_source.parameters)
-        query = nosql_apply_parameters_to_query(data_source.query, data_source.parameters)
-        limit = float('inf') if data_source.limit == -1 else data_source.limit
+            if len(variable_data) > 0:
+                for data in variable_data:
+                    variable_response_list += list(
+                        map(
+                            lambda obj: DICTIONARY_OF_FORMATTERS.get(dataset, 'users')(obj),
+                            data.get(dataset),
+                        )
+                    )
+            return team_response_list, variable_response_list
 
-        current_page = 1
-        is_last_page = False
-        data = []
+    async def _get_tags(self, dataset: str, limit) -> List[dict]:
+        """Triggers fetches for tags and does preliminary filtering process"""
+        headers = {'Authorization': BEARER_API_KEY, 'Bearer-Auth-Id': self.bearer_auth_id}
+        async with ClientSession(headers=headers) as session:
+            raw_data = await fetch_page(dataset, [], session, limit, 1,)
+            tags_data_list = []
+            if len(raw_data) > 0:
+                for data in raw_data:
+                    tags_data_list += data.get('tags')
+            return tags_data_list
```

Comment:
What about:

```suggestion
        tags_data_list = []
        for data in raw_data:
            tags_data_list += data['tags']
        return tags_data_list
```

This could even be written in a oneliner but I'm not a huge fan:

```python
return sum([data['tags'] for data in raw_data], [])
```